// shared/vitest.config.ts
// vitest.config.ts - Root configuration for monorepo
import { defineConfig } from 'vitest/config';
import tsconfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    // This is the default configuration for all tests
    globals: true,
    exclude: [
      '/node_modules/', 
      '/dist/', 
      '/e2e/',
      '/checkly/',  // Exclude all Checkly tests from unit test runs
      '/apps/web/public/build/',
      '/apps/web/build/',
      // Exclude problematic virtual files generated by Remix/Vite
      '/build/server-assets-manifest:*',
      '/build/server-entry-module:*',
      '/@remix-run/',
      // Exclude all virtual files with special characters
      '/@remix-run/*',
      '/:',
      '/checkly.config.ts',
      '/playwright.config.ts',
    ],
    reporters: ['default', ['junit', { outputFile: 'artifacts/junit/unit-results.xml' }]],
    coverage: {
      provider: 'v8',
      reporter: ['text', 'lcov'],
      reportsDirectory: 'artifacts/coverage',
      // Exclude problematic paths and focus on source code
      exclude: [
        '/node_modules/',
        '/dist/',
        '/e2e/',
        '/apps/web/public/build/',
        '/apps/web/build/',
        // Exclude problematic virtual files generated by Remix/Vite
        '/build/server-assets-manifest:*',
        '/build/server-entry-module:*',
        '/@remix-run/',
        // Exclude all virtual files with special characters
        '/@remix-run/*',
        '/:',
        '/checkly.config.ts',
        '/playwright.config.ts',
        // Only include actual source files
        '!packages/shared/src/',
        '!apps/api/src/',
        '!apps/web/app/',
      ],
    },
  },
});
